volatility3/framework/plugins/linux/graphics/fbdev.py

```diff
@@ -0,0 +1,314 @@
+        # As Volatility3 is not a video format converter, it is best to play it safe and let the user parse
+        # the raw data manually (with ffmpeg for example).
+        elif fb_info.var.grayscale > 1:
+            fourcc = linux.LinuxUtilities.convert_fourcc_code(fb_info.var.grayscale)
+            warn_msg = f"""Framebuffer "{id}" uses a FOURCC pixel format "{fourcc}" that isn't natively supported.
+You can try using ffmpeg to decode the raw buffer. Example usage:
+"ffmpeg -pix_fmts" to list supported formats, then
+"ffmpeg -f rawvideo -video_size {fb_info.var.xres_virtual}x{fb_info.var.yres_virtual} -i <FILENAME>.raw -pix_fmt <FORMAT> output.png"."""
+            vollog.warning(warn_msg)
+
+        # Prefer using the virtual resolution, instead of the visible one.
+        # This prevents missing non-visible data stored in the framebuffer.
+        fb = Framebuffer(
+            id,
+            xres_virtual=fb_info.var.xres_virtual,
+            yres_virtual=fb_info.var.yres_virtual,
+            line_length=fb_info.fix.line_length,
+            bpp=fb_info.var.bits_per_pixel,
+            size=fb_info.var.yres_virtual * fb_info.fix.line_length,
+            color_fields=color_fields,
+            fb_info=fb_info,
+        )
+
+        return fb
+
+    def _generator(self):
+        kernel_name = self.config["kernel"]
+        kernel = self.context.modules[kernel_name]
+
+        if not kernel.has_symbol("num_registered_fb"):
+            raise exceptions.SymbolError(
+                "num_registered_fb",
+                kernel.symbol_table_name,
+                "The provided symbol does not exist in the symbol table. This means you are either analyzing an unsupported kernel version or that your symbol table is corrupt.",
+            )
+
+        num_registered_fb = kernel.object_from_symbol("num_registered_fb")
+        if num_registered_fb < 1:
+            vollog.info("No registered framebuffer in the fbdev API.")
+            return None
+
+        registered_fb = kernel.object_from_symbol("registered_fb")
+        fb_info_list = utility.array_of_pointers(
+            registered_fb,
+            num_registered_fb,
+            kernel.symbol_table_name + constants.BANG + "fb_info",
+            self.context,
+        )
+
+        for fb_info in fb_info_list:
+            fb = self.parse_fb_info(fb_info)
+            file_output = "Disabled"
+            if self.config["dump"]:
+                try:
+                    file_output = self.dump_fb(
+                        self.context, kernel_name, self.open, fb, bool(fb.color_fields)
+                    )
+                except exceptions.InvalidAddressException as excp:
+                    vollog.error(
+                        f'Layer {excp.layer_name} failed to read address {hex(excp.invalid_address)} when dumping framebuffer "{fb.id}".'
+                    )
+                    file_output = "Error"
```

Comment:
This is the current (manual) way of handling errors in file outputting across the framework, but indeed a `OutputFailure()` exception could be added somewhere in the renderers. I made sure to generate an error beforehand to inform the user on the failure reason, and not overload the `Filename` column. 

I can open an issue to track it and eventually update related bits once and for all when a solution comes out ?

Reply:
Well, I don't necessarily want to throw an exception (which then stops the rest of the run), but I think the string "Error" isn't how we should indicate this having been an error, I think it should be a BaseAbsentValue derived value.  So either `Unreadable`, `Unparseable`, `NotAvailable` or we could invent a new one?

Reply:
Yes, I meant a new renderer not an exception sorry.

`UnreadableValue()` outputs a single dash (`-`) as well, which is fine if we want to rely on the error message solely ?

Reply:
It does in the CLI, but the point is that a web interface may display that return type differently.  If the plugin chooses that it's "Error", then it's Error for everyone, if the plugin returns it as a `BaseAbsentValue` then it might be able to do something better/cleverer that we haven't though of (and if not it can always just put a `-`).  I think the column data isn't the place to report errors, the error framework we've got is how that should be done, so yeah, I'd be fine just returning a BaseAbsentValue (or making a new one that we could provide an explanation with, but the CLI representation could just be "Error")?  Essentially, we should be programmatically providing the information about the error, rather than in a text representation...  Throwing an error should stop the program output anyway, so it might be academic?  5:P

Reply:
(As in, you might want to make it `vollog.warn` rather than `vollog.error`, since `error` will terminate the program afterwards)...  5;)

Reply:
Understandable, I'll make it an `UnreadableValue()`.

`vollog.error` does not exit the program, as it is just logging ? See this:

```sh
vollog.error("error")
vollog.info("still here")
```

I think it is more adapted in this case, as it is truly an error.

Reply:
Oh, I thought any time you logged an error it also caused a `sys.exit`?  Fair enough, perhaps I was thinking of something else?

Reply:
Sure, if it doesn't kill the program I'm fine with logging it as an error.